app/lib/multiple_parameters_date_type.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class MultipleParametersDateType < ActiveModel::Type::Value
+  class PotentialDate
+    include ActiveModel::Model
+    attr_accessor :year, :month, :day
+  end
+
+  def self.process(params)
+    params.keys.select { |key| key.to_s.match?(/\(\d+i\)$/) }.map do |key|
+      attribute_name = key.to_s.split('(').first
+
+      year = params.delete("#{attribute_name}(1i)")
+      month = params.delete("#{attribute_name}(2i)")
+      day = params.delete("#{attribute_name}(3i)")
+
+      params[attribute_name] = PotentialDate.new(year:, month:, day:) if year.present? && month.present? && day.present?
+    end
+
+    params
+  end
+
+  def cast(value)
+    if value.is_a?(PotentialDate)
+      begin
+        Date.new(value.year.to_i, value.month.to_i, value.day.to_i)
+      rescue ArgumentError
+        nil
+      end
+    else
+      super
+    end
```

Comment:
I consciously chosen not to use the active model date type as have a simple version.